rskj-core/src/main/java/co/rsk/rpc/Web3RskImpl.java

```diff
@@ -172,4 +187,36 @@ private String toSmallHash(byte[] input) {
         return Hex.toHexString(input).substring(56,64);
     }
 
+    private BtcBlock getBtcBlock(String blockHeaderHex, NetworkParameters params) {
+        byte[] bitcoinBlockByteArray = Hex.decode(blockHeaderHex);
+        return params.getDefaultSerializer().makeBlock(bitcoinBlockByteArray);
+    }
+
+    private String extractBlockHashForMergedMining(BtcTransaction coinbase) {
+        byte[] coinbaseAsByteArray = coinbase.bitcoinSerialize();
+        List<Byte> coinbaseAsByteList = Arrays.asList(ArrayUtils.toObject(coinbaseAsByteArray));
+
+        List<Byte> rskTagAsByteList = Arrays.asList(ArrayUtils.toObject(RskMiningConstants.RSK_TAG));
+
+        int rskTagPosition = Collections.lastIndexOfSubList(coinbaseAsByteList, rskTagAsByteList);
+        byte[] blockHashForMergedMiningArray = new byte[Keccak256Helper.Size.S256.getValue()/8];
+        System.arraycopy(coinbaseAsByteArray, rskTagPosition+ RskMiningConstants.RSK_TAG.length, blockHashForMergedMiningArray, 0, blockHashForMergedMiningArray.length);
+        return TypeConverter.toJsonHex(blockHashForMergedMiningArray);
+    }
+
+    private List<String> parseTransactionHashes(String txnHashesHex) {
+        List<String> txnHashes = new ArrayList<>();
+        for (int start = 0; start < txnHashesHex.length(); start += 64) {
```

Comment:
Please create a constant for `64`

Reply:
Since a last minute optimization this code no longer exists